src/main/java/org/opentripplanner/routing/api/request/preference/RoutingPreferences.java

```diff
@@ -308,11 +317,21 @@ public Builder withItineraryFilter(Consumer<ItineraryFilterPreferences.Builder>
       return this;
     }
 
+    public MappingPreferences mappingPreferences() {
+      return mappingPreferences == null ? original.mapping : mappingPreferences;
+    }
+
     public Builder apply(Consumer<Builder> body) {
       body.accept(this);
       return this;
     }
 
+    public Builder withMapping(Consumer<MappingPreferences.Builder> body) {
+      this.mappingPreferences =
+        ifNotNull(this.mappingPreferences, original.mapping).copyOf().apply(body).build();
```

Comment:
I guess we don't need to create a copy if the mapping preferences are null since we are just using the original?